tests/quarkus/src/test/java/io/hawt/tests/quarkus/suites/CucumberQuarkusTest.java

```diff
@@ -0,0 +1,239 @@
+
+        final ExitStatus exitStatus = new ExitStatus(runtimeOptions);
+        plugins.addPlugin(exitStatus);
+        if (runtimeOptions.isMultiThreaded()) {
+            plugins.setSerialEventBusOnEventListenerPlugins(eventBus);
+        } else {
+            plugins.setEventBusOnEventListenerPlugins(eventBus);
+        }
+        ObjectFactory objectFactory = new io.quarkiverse.cucumber.CucumberQuarkusTest.CdiObjectFactory();
+
+        ObjectFactorySupplier objectFactorySupplier = () -> objectFactory;
+
+        Runner runner = new Runner(eventBus,
+            Collections.singleton(new JavaBackendProviderService().create(objectFactorySupplier.get(),
+                objectFactorySupplier.get(),
+                () -> Thread.currentThread()
+                    .getContextClassLoader())),
+            objectFactorySupplier.get(),
+            runtimeOptions);
+
+        CucumberExecutionContext context = new CucumberExecutionContext(eventBus, exitStatus, () -> runner);
+
+        List<DynamicNode> features = new LinkedList<>();
+        features.add(DynamicTest.dynamicTest("Start Cucumber", context::startTestRun));
+
+        Predicate<Pickle> filters = new Filters(runtimeOptions);
+
+        featureSupplier.get().forEach(f -> {
+            List<DynamicTest> tests = new LinkedList<>();
+            tests.add(DynamicTest.dynamicTest("Start Feature", () -> context.beforeFeature(f)));
+            f.getPickles()
+                .stream()
+                .filter(filters)
+                .forEach(p -> tests.add(DynamicTest.dynamicTest(p.getName(), () -> {
+                    AtomicReference<TestStepFinished> resultAtomicReference = new AtomicReference<>();
+                    EventHandler<TestStepFinished> handler = event -> {
+                        if (event.getResult().getStatus() != Status.PASSED) {
+                            // save the first failed test step, so that we can get the line number of the cucumber file
+                            resultAtomicReference.compareAndSet(null, event);
+                        }
+                    };
+
+                    try (TestCaseResultObserver resultObserver = new TestCaseResultObserver(eventBus)) {
+
+                        context.runTestCase(r -> r.runPickle(p));
+                        resultObserver.assertTestCasePassed(TestSkippedException::new,
+                            (t) -> new TestSkippedException("", t),
+                            UndefinedStepException::new, RuntimeException::new
+                        );
+                    }
+                })));
+
+            if (tests.size() > 1) {
+                features.add(DynamicContainer.dynamicContainer(f.getName().orElse(f.getSource()), tests.stream()));
+            }
+        });
+
+        features.add(DynamicTest.dynamicTest("Finish Cucumber", context::finishTestRun));
+
+        return features;
+    }
+
+    public static class CdiObjectFactory implements ObjectFactory {
```

Comment:
Used to bridge Quarkus CDI and Cucumber - here's the main breaking change between Quarkus 2.x and Quarkus 3.x so this needed different imports `javax` -> `jakarta`